src/layers/generic.cpp

```diff
@@ -408,17 +411,26 @@ namespace marian {
     std::string name = opt<std::string>("prefix");
     int dimVoc = opt<int>("dimVocab");
     int dimEmb = opt<int>("dimEmb");
+    int dimFactorEmb = opt<int>("dimFactorEmb");
+    std::string factorCombine = opt<std::string>("factorsCombine");
```

Comment:
Try to avoid string copies if you can.  (Yes I know the above code is guilty of this.)

Reply:
removed the string copy